src/manager.cpp

```diff
@@ -221,123 +221,33 @@ int Manager::updateKeyword(const types::Path i_vpdPath,
         return -1;
     }
 
+    types::Path l_fruPath;
     nlohmann::json l_sysCfgJsonObj{};
 
     if (m_worker.get() != nullptr)
     {
         l_sysCfgJsonObj = m_worker->getSysCfgJsonObj();
-    }
-
-    types::Path l_fruPath = i_vpdPath;
-    types::Path l_inventoryObjPath;
-    types::Path l_redundantFruPath;
-
-    if (!l_sysCfgJsonObj.empty())
-    {
         try
         {
-            // Get hardware path from system config JSON.
-            const types::Path& l_tempPath =
-                jsonUtility::getFruPathFromJson(l_sysCfgJsonObj, i_vpdPath);
-
-            if (!l_tempPath.empty())
-            {
-                // Save the FRU path to update on hardware
-                l_fruPath = l_tempPath;
-
-                // Get inventory object path from system config JSON
-                l_inventoryObjPath = jsonUtility::getInventoryObjPathFromJson(
-                    l_sysCfgJsonObj, i_vpdPath);
-
-                // Get redundant hardware path if present in system config JSON
-                l_redundantFruPath =
-                    jsonUtility::getRedundantEepromPathFromJson(l_sysCfgJsonObj,
-                                                                i_vpdPath);
-            }
+            // Get the EEPROM path
+            l_fruPath = jsonUtility::getFruPathFromJson(l_sysCfgJsonObj,
+                                                        i_vpdPath);
         }
-        catch (const std::exception& e)
+        catch (const std::exception& l_exception)
         {
-            return -1;
+            // Do nothing in case of inventory path or the system config JSON is
+            // empty.
         }
     }
 
-    // Update keyword's value on hardware
-    int l_bytesUpdatedOnHardware = updateKeywordOnHardware(
-        l_fruPath, l_sysCfgJsonObj, i_paramsToWriteData);
-
-    if (l_bytesUpdatedOnHardware == -1)
+    if (l_fruPath.empty())
```

Comment:
If "l_fruPath" is empty till this point then the code will always end up with throwing exception if the path is not EEPROM path outside JSON. Implies this flow will not work if the path is inventory path. So i case we are not supporting inventory path as parameter, should the parameter named be changed?